x/logic/interpreter/fs/fs.go

```diff
@@ -0,0 +1,99 @@
+package fs
+
+import (
+	goctx "context"
+	"io"
+	"io/fs"
+	"time"
+)
+
+// FileSystem is the custom virtual file system used into the blockchain.
+// It will hold a list of handler that can resolve file URI and return the corresponding binary file.
+type FileSystem struct {
+	ctx    goctx.Context
+	router Router
+}
+
+// New return a new FileSystem object that will handle all virtual file on the interpreter.
+// File can be provided from different sources like CosmWasm cw-storage smart contract.
+func New(ctx goctx.Context, handlers []URIHandler) FileSystem {
+	router := NewRouter()
+	for _, handler := range handlers {
+		router.RegisterHandler(handler)
+	}
+	return FileSystem{
+		ctx:    ctx,
+		router: router,
+	}
+}
+
+// Open will read the entire file from ReadFile interface,
+// Since file is provided by a provider that do not support streams.
+func (f FileSystem) Open(name string) (fs.File, error) {
+	data, err := f.ReadFile(name)
+	if err != nil {
+		return nil, &fs.PathError{
+			Op:   "open",
+			Path: name,
+			Err:  err,
+		}
+	}
+	return Object(data), nil
+}
+
+// ReadFile read the entire file at the uri provided.
+// Parse all handler and return the first supported handler file response.
+func (f FileSystem) ReadFile(name string) ([]byte, error) {
+	return f.router.Open(f.ctx, name)
+}
+
+type Object []byte
+
+type ObjectInfo struct {
```

Comment:
I don't clearly see the relation with `fs.File` and `fs.FileInfo`.

Reply:
@ccamel  At time of your review, this Object has been moved into another file, call Object yet but the type is now a struct that hold a buffer and some virtual file informations. 
This object implement `fs.File`. ans `fs.FileInfo` interface that respond to the requirements of the `fs.FS` `Open` interface. 
Maybe is the name `Object` that is not good, we can rename it like `VirtualFile` maybe ?

Reply:
`VirtualFile` would be better in my opinion. Ideally, I would split the structure into two, one for the `VirtualFile` and one for the `VirtualFileInfo`, but that's a matter of taste.

Reply:
I agree with the `VirtualFile` and `VirtualFileInfo` naming.